cf_units/tests/unit/unit/test_as_unit.py

```diff
@@ -0,0 +1,93 @@
+# (C) British Crown Copyright 2014, Met Office
+#
+# This file is part of cf_units.
+#
+# cf_units is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cf_units is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cf_units.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the `cf_units.as_unit` function."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+import copy
+import unittest
+
+import cf_units
+from cf_units import as_unit, Unit
+
+
+class StubUnit(object):
+    def __init__(self, calendar=None):
+        self._calendar = str(calendar) if calendar is not None else None
+
+    def __str__(self):
+        return self.origin
+
+    @property
+    def origin(self):
+        return 'hours since 1970-01-01 00:00:00'
+
+    @property
+    def ut_unit(self):
+        return 22943184
```

Comment:
In general, are we able to remove all of these properties and make them attributes?#

> Wait I know the explanation for that!

:laughing: